pkg/parser/parser_csv.go

```diff
@@ -0,0 +1,77 @@
+package parser
+
+import (
+	"bytes"
+	"encoding/csv"
+	"errors"
+	"fmt"
+	"io"
+)
+
+type CsvParser struct {
+}
+
+func (x CsvParser) Convert(unprocessed []byte) (doc Document, err error) {
+	d := Document{
+		Attributes: Attributes{
+			Columns: make([]string, 0, 0),
+			Values:  make(map[string][]string),
+		},
+	}
+
+	defer func() (Document, error) {
+		if r := recover(); r != nil {
+			return d, errors.New(fmt.Sprintf("%s", r))
+		}
+		return d, err
```

Comment:
This (I believe) won't work correctly, if you want it to change the return value of the function, you'll need to reference the parent function returned parameters